src/session/sidebar/row.rs

```diff
@@ -127,6 +128,10 @@ impl Row {
 
         let self_ = imp::Row::from_instance(self);
 
+        while let Some(binding) = self_.bindings.borrow_mut().pop() {
```

Comment:
I would suggest saving a `RefMut<'a, Vec<_>>` in a variable before this `while let` by doing a
```rust
let mut bindings = self_.bindings.borrow_mut();
```
Later you could just use this `RefMut` and don't need to strees the runtime borrow checker unnecessarily.